### DISCUSSION TITLE
Alternative way to use className in typescript

### DISCUSSION THREAD

**Discussion by mateusz-ramski** [nativewind/nativewind#283] [Show and tell]
Hi, with your typescript solution, in my custom components i should always extend my props interface with one of "base" interfaces like ViewProps, TextProps... But im not aways passing all props to my custom component, so it can lead to wrong intellisense suggestions.

You can use className in all components just by adding to app.d.ts:

```
declare namespace JSX {
    interface IntrinsicAttributes {
        className?: string
    }
}
```

Great project! I will use it in our production app.

**Comment by marklawlor** [comment]
Hi, thanks for this. I'll include this in the next major version of NativeWind. I'm going to close to issue and convert it to a showcase discussion

**Comment by chmieleski** [comment]
The only problem I see using `IntrinsicAttributes` is that sometimes components we build may not have styling capabilities, which can lead the developers to errors.

**Reply by marklawlor** [reply]
I considered that. Basically I'm stuck between being too open, and causing possible confusion, and being too closed and forcing developers to extend types :man_shrugging: 

I think extending `IntrinsicAttributes` is the better of the two evils. There might be some Typescript infer magic we can use to only apply `className` when an interface as a `style` attribute but I don't have time to explore that right now. I'm open to PR suggestions

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
